backend/src/apiserver/server/experiment_server.go

```diff
@@ -163,18 +214,67 @@ func (s *ExperimentServer) ListExperimentsV1(ctx context.Context, request *api.L
 	if err != nil {
 		return nil, util.Wrap(err, "List experiments failed.")
 	}
-	return &api.ListExperimentsResponse{
+	return &apiv1beta1.ListExperimentsResponse{
+			Experiments:   ToApiExperimentsV1(experiments),
+			TotalSize:     int32(total_size),
+			NextPageToken: nextPageToken},
+		nil
+}
+
+func (s *ExperimentServer) ListExperiments(ctx context.Context, request *apiv2beta1.ListExperimentsRequest) (
+	*apiv2beta1.ListExperimentsResponse, error) {
+	if s.options.CollectMetrics {
+		listExperimentsV1Requests.Inc()
+	}
+
+	opts, err := validatedListOptions(&model.Experiment{}, request.PageToken, int(request.PageSize), request.SortBy, request.Filter)
```

Comment:
Thank you for pointing out this! I didn't pay attention to the inside of this function. You are right, and I'm surprised that this PR worked when I was testing it. After looking deeper into the code, I think it worked because although the whole file used `api/v1beta1`, the types used in `validatelistOptions` are interfaces. We can easily keep this part when we deprecate v1.